assets/backport/github/github.go

```diff
@@ -0,0 +1,336 @@
+	return newBranchName, nil
+}
+
+// CreatePullRequest creates a pull request.
+func (c *Client) CreatePullRequest(ctx context.Context, baseBranch string, headBranch string, titleAndBody string) error {
+	newPR := &go_github.NewPullRequest{
+		Title:               go_github.String(titleAndBody),
+		Head:                go_github.String(headBranch),
+		Base:                go_github.String(baseBranch),
+		Body:                go_github.String(titleAndBody),
+		MaintainerCanModify: go_github.Bool(true),
+	}
+	_, _, err := c.Client.PullRequests.Create(ctx, c.Organization, c.Repository, newPR)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetPullRequestMetadata gets the commit shas, title, and body for a pull request
+// associated with the passed in branch name.
+func (c *Client) GetPullRequestMetadata(ctx context.Context, number int) (commits []string, branchName string, err error) {
+	pull, _, err := c.Client.PullRequests.Get(ctx, c.Organization, c.Repository, number)
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	if pull.GetState() != backportPRState {
+		return nil, "", trace.Errorf("pull request %v is not closed", number)
+	}
+	if strings.TrimPrefix(pull.GetBase().GetRef(), branchRefPrefix) != masterBranchName {
+		return nil, "", trace.Errorf("pull request %v's base is not master", number)
+	}
+
+	commits, err = c.getPullRequestCommits(ctx, pull.GetNumber())
+	if err != nil {
+		return nil, "", trace.Wrap(err)
+	}
+	return commits, strings.TrimPrefix(pull.GetHead().GetRef(), branchRefPrefix), nil
+}
+
+// cherryPickCommitsOnBranch cherry picks a list of commits on a given branch.
+func (c *Client) cherryPickCommitsOnBranch(ctx context.Context, branchName string, commits []string) error {
+	branch, _, err := c.Client.Repositories.GetBranch(ctx, c.Organization, c.Repository, branchName, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Get the branch's HEAD.
+	headCommit, _, err := c.Client.Git.GetCommit(ctx,
+		c.Organization,
+		c.Repository,
+		branch.GetCommit().GetSHA())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for i := 0; i < len(commits); i++ {
+		cherryCommit, _, err := c.Client.Git.GetCommit(ctx, c.Organization, c.Repository, commits[i])
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		tree, sha, err := c.cherryPickCommit(ctx, branchName, cherryCommit, headCommit)
+		if err != nil {
+			defer func() {
+				refName := fmt.Sprintf("%s%s", branchRefPrefix, branchName)
+				c.Client.Git.DeleteRef(ctx, c.Organization, c.Repository, refName)
+			}()
+			return trace.Wrap(err)
+		}
+		headCommit.SHA = &sha
+		headCommit.Tree = tree
+	}
+	return nil
+}
+
+// cherryPickCommit cherry picks a single commit on a branch.
+func (c *Client) cherryPickCommit(ctx context.Context, branchName string, cherryCommit, headBranchCommit *go_github.Commit) (*go_github.Tree, string, error) {
+	if len(cherryCommit.Parents) != 1 {
+		return nil, "", trace.BadParameter("merge commits are not supported")
```

Comment:
What does this mean? If someone hits the `Update Branch` button in the UI, will this tool fail?

Reply:
Yes because updating the branch results in a merge commit.

Reply:
Why not skip over these commits instead of a hard fail?

Reply:
I suppose we can do that. I will update.